SQLQueryGroupProjectTeamB1Julia.sql

```diff
@@ -0,0 +1,273 @@
+--Julia Shull
+--Group Project Code
+--November 2019
+
+USE Proj_B1
+
+--STORED PROCEDURES
+
+--Write the stored procedure to create a new shift.
+
+CREATE PROCEDURE SP_NewShift
+@QuarterName VARCHAR(10),
+@MonthNumber INT,
+@DayNumber INT,
+@BeginHour INT,
+@EndHour INT,
+@Location VARCHAR(50),
+@Compensation VARCHAR(50),
+@ShiftTypeName VARCHAR(50),
+@Year VARCHAR(4)
+
+AS
+
+DECLARE @S_ID INT, @LOC_ID INT, @Q_ID INT, @MON_ID INT, @DAY_ID INT, @BH_ID INT, @EH_ID INT, @COMP_ID INT, @ST_ID INT
+
+SET @S_ID = (SELECT S.ShiftID
+			 FROM tblSHIFT S
+				JOIN tblQuarter Q
+					ON Q.QuarterID = S.QuarterID
+				JOIN tblMONTH M
+					ON M.MonthID = S.MonthID
+				JOIN tblDAY D
+					ON D.DayID = S.ShiftID
+				JOIN tblHOUR H
+					ON H.HourID = S.BeginHourID
+				JOIN tblCOMPENSATION C
+					ON C.CompID = S.CompID
+				JOIN tblSHIFT_TYPE ST
+					ON ST.ShiftTypeID = S.ShiftTypeID
+				JOIN tblLOCATION L
+					ON L.LocationID = S.LocationID
+			 WHERE  Q.QuartName = @QuarterName
+				AND M.MonthNum = @MonthNumber
+				AND D.[DayName] = @DayNumber
+				AND S.BeginHourID = @BeginHour
+				AND S.EndHourID = @EndHour
+				AND L.LocationName = @Location
+				AND C.CompName = @Compensation
+				AND ST.ShiftTypeName = @ShiftTypeName
+				AND S.[YEAR] = @Year)
+
+SET @LOC_ID = (SELECT L.LocationID
+			   FROM tblLOCATION L
+			   WHERE L.LocationName = @Location)
+SET @Q_ID = (SELECT Q.QuarterID
+			 FROM tblQUARTER Q
+			 WHERE Q.QuartName = @QuarterName)
+SET @MON_ID = (SELECT M.MonthID
+			   FROM tblMONTH M
+			   WHERE M.MonthNum = @MonthNumber)
+SET @DAY_ID = (SELECT D.DayID
+			   FROM tblDAY D
+			   WHERE D.[DayName] = @DayNumber)
+SET @BH_ID = (SELECT H1.HourID
+			  FROM tblHOUR H1
+			  WHERE H1.HourName = @BeginHour)
+SET @EH_ID = (SELECT H2.HourID
+			  FROM tblHOUR H2
+			  WHERE H2.HourName = @EndHour)
+SET @COMP_ID = (SELECT C.CompID
+				FROM tblCOMPENSATION C
+				WHERE C.CompName = @Compensation)
+SET @ST_ID = (SELECT ST.ShiftTypeId
+			  FROM tblSHIFT_TYPE ST
+			  WHERE ST.ShiftTypeName = @ShiftTypeName)
+
+BEGIN TRANSACTION JT1
+INSERT INTO tblSHIFT(ShiftID, LocationID, QuarterID, MonthID, DayID, [YEAR], BeginHourID, EndHourID, CompID, ShiftTypeID)
+VALUES (@S_ID, @LOC_ID, @Q_ID, @MON_ID, @DAY_ID, @Year, @BH_ID, @EH_ID, @COMP_ID, @ST_ID)
+END TRANSACTION JT1

```

Comment:
Instead of End transaction, you have to replace with this to get the transaction thing working properly
IF @@ERROR <> 0
	ROLLBACK TRAN JT1
ELSE
	COMMIT TRAN JT1